ci-operator/step-registry/azure/provision/vault-key/azure-provision-vault-key-commands.sh

```diff
@@ -11,24 +28,34 @@ az --version
 az cloud set --name AzureCloud
 az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
 
+set -x
+
 KV_BASE_NAME="${NAMESPACE}-${UNIQUE_HASH}"
 LOCATION="$LEASED_RESOURCE"
-RESOURCE_GROUP="$(<"${SHARED_DIR}/resourcegroup")"
+if [[ $AZURE_KEYVAULT_USE_AKS_RG == "true" ]]; then
+    RESOURCE_GROUP="$(<"${SHARED_DIR}/resourcegroup_aks")"
+else
+    RESOURCE_GROUP="$(<"${SHARED_DIR}/resourcegroup")"
+fi
 az group show --name "$RESOURCE_GROUP"
 
 echo "Creating KeyVault"
 KEYVAULT_NAME="${KV_BASE_NAME}-kv"
 # Set soft delete data retention to 7 days (minimum) instead of the default 90 days to reduce the risk of Key Vault name collisions
-az keyvault create -n "$KEYVAULT_NAME" -g "$RESOURCE_GROUP" -l "$LOCATION" --enable-purge-protection true --enable-rbac-authorization false --retention-days 7
+az keyvault create -n "$KEYVAULT_NAME" -g "$RESOURCE_GROUP" -l "$LOCATION" --enable-purge-protection --enable-rbac-authorization --retention-days 7
 
 echo "Granting ServicePrincipal permissions to the KeyVault"
-az keyvault set-policy -n "$KEYVAULT_NAME" --key-permissions create decrypt encrypt get --spn "$AZURE_AUTH_CLIENT_ID"
+SP_ID=$(az ad sp show --id "$AZURE_AUTH_CLIENT_ID" --query id -o tsv)
+KV_ID=$(az keyvault show --name "$KEYVAULT_NAME" -g "$RESOURCE_GROUP" --query id -o tsv)
+az role assignment create --assignee "$SP_ID" --scope "$KV_ID" --role "Key Vault Administrator"
```

Comment:
Shall we need to deprovision the assignment when destroying the cluster ? cc @jinyunma to help confirm ?

Reply:
It'd not needed imo. Ref. https://learn.microsoft.com/en-us/answers/questions/1657544/need-to-remove-rbac-role-when-try-to-delete-resour

Reply:
The linked doc is about resource group, here the assignment is targeted for SP, and the SP would never be destroyed. Can you figure out some command lines to identify if the assignment is still existing or not after your ci job get completed.

Reply:
In the document I referenced earlier, the assignment is made to a user, group, or service principal (the assignee can only be one of these three types) and scoped to a resource group, which is later deleted. In my case, the assignment is to a service principal and scoped to a Key Vault, which is then deleted. These two scenarios are analogous. 

For completeness, I also conducted the following test:
```bash
az keyvault create -n "fxie-test-2" -g "os4-common" -l "eastus" --enable-purge-protection --enable-rbac-authorization --retention-days 7

az role assignment create --assignee XXX --scope "/subscriptions/XXX/resourceGroups/os4-common/providers/Microsoft.KeyVault/vaults/fxie-test-2" --role "Key Vault Administrator"

az role assignment list --scope "/subscriptions/XXX/resourceGroups/os4-common/providers/Microsoft.KeyVault/vaults/fxie-test-2" --output table
Principal                             Role                     Scope
------------------------------------  -----------------------  -----------------------------------------------------------------------------------------------------------------------------
dfdd56ef-0a6b-49a4-bcee-4aad2eca18b0  Key Vault Administrator  /subscriptions/XXX/resourceGroups/os4-common/providers/Microsoft.KeyVault/vaults/fxie-test-2

az keyvault delete --name "fxie-test-2" --resource-group "os4-common"

az role assignment list --scope "/subscriptions/XXX/resourceGroups/os4-common/providers/Microsoft.KeyVault/vaults/fxie-test-2" --output table
(ResourceNotFound) The Resource 'Microsoft.KeyVault/vaults/fxie-test-2' under resource group 'os4-common' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
Code: ResourceNotFound
Message: The Resource 'Microsoft.KeyVault/vaults/fxie-test-2' under resource group 'os4-common' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix
```

Reply:
Nice, thanks for your checking.

Reply:
since the scope of this assignment is on KeyVault, we don't need to delete assignment separately because related KeyVault will be destroyed in the deprovision stage.
We only care about the assignment on the scope of subscription.

Reply:
Thanks, good to know.